File: src/Players/HumanPlayer.cpp
#include "HumanPlayer.hpp"

HumanPlayer::HumanPlayer() : pending(false), pendingMove() {
}

bool HumanPlayer::isHuman() const {
	return true;
}

Move HumanPlayer::chooseMove(const GameState&, const Rules&) {
	return Move();
}

void HumanPlayer::setPendingMove(const Move& move) {
	pendingMove = move;
	pending = true;
}

bool HumanPlayer::hasPendingMove() const {
	return pending;
}

Move HumanPlayer::takePendingMove() {
	pending = false;
	return pendingMove;
}

File: backend/human_player.go
package main

type HumanPlayer struct {
	pending     bool
	pendingMove Move
}

func NewHumanPlayer() *HumanPlayer {
	return &HumanPlayer{}
}

func (h *HumanPlayer) IsHuman() bool {
	return true
}

func (h *HumanPlayer) ChooseMove(GameState, Rules) Move {
	return Move{}
}

func (h *HumanPlayer) SetPendingMove(move Move) {
	h.pendingMove = move
	h.pending = true
}

func (h *HumanPlayer) HasPendingMove() bool {
	return h.pending
}

func (h *HumanPlayer) TakePendingMove() Move {
	h.pending = false
	return h.pendingMove
}
